### PR TITLE
fix: Fixes issue #2579

### PR DESCRIPTION
**Pull request by jleinenbach**
### Summary of the CPU Issue and Solution

**Problem:**  
The implementation with **parallel loading** of Alexa components caused high CPU usage. Multiple components were loaded simultaneously, which overwhelmed the system due to the concurrent resource-heavy operations.

**Solution:**
1. **Serial Loading:**   I addressed the CPU problem by **serializing** the loading process. This means components are now loaded **one after the other** instead of all at once, reducing the CPU load.
   
2. **Condition with `entry_setup`:**   I reintroduced the check for `if not entry_setup`. This ensures that we only initialize components if they are not already set up, preventing redundant operations. For components already initialized, the same `async_forward_entry_setup` method is called to reload them.

**Outcome:**  
By loading components sequentially and ensuring that unnecessary setups are skipped, we significantly reduce the CPU load while maintaining efficient processing of new Alexa clients.

### CODE REVIEW

**Review by MichaelMKKelly — CHANGES_REQUESTED** [review]
line 1541 needs a ) removed from the end to fix a syntax error

File: custom_components/alexa_media/__init__.py [hunk 1]
```diff
@@ -1343,63 +1430,171 @@ async def http2_error_handler(message):
...
-        _LOGGER.debug("Removing config_flows data")
-        async_dismiss_persistent_notification(
-            hass, f"alexa_media_{slugify(email)}{slugify((entry.data['url'])[7:])}"
-        )
-        hass.data[DATA_ALEXAMEDIA].pop("config_flows")
-    if not hass.data[DATA_ALEXAMEDIA]:
-        _LOGGER.debug("Removing alexa_media data structure")
-        if hass.data.get(DATA_ALEXAMEDIA):
-            hass.data.pop(DATA_ALEXAMEDIA)
+                    _LOGGER.warning("Config flow '%s' not found for %s", flow_id, hide_email(email)))
```
Comment (apbarratt) [inline custom_components/alexa_media/__init__.py]:
```suggestion
                    _LOGGER.warning("Config flow '%s' not found for %s", flow_id, hide_email(email))
```

Comment (apbarratt) [reply custom_components/alexa_media/__init__.py]: As flagged by @MichaelMKKelly

**Comment by alandtse** [comment]
Please note, while I'm not opposed to what looks like ChatGPT changes, given the recent issue with #2579, I'm going to require someone actually test it before merging.

That said, I want to thank you for being willing to do PRs.

**Comment by danielbrunt57** [comment]
@jleinenbach Are you aware of [PR #2589](https://github.com/alandtse/alexa_media_player/pull/2580) to address memory exhaustion/infinite loop?

**Comment by danielbrunt57** [comment]
@alandtse As you're no doubt aware, I implemented @jleinenbach's original code and then amended the PR with 3 edits to address issues I encountered with it.  That said, I'm running the resulting `__init__.py` in my system and everything here is working fine.  
Also, I just now implemented the `helpers.py` code change in PR [#2589](https://github.com/alandtse/alexa_media_player/pull/2580) and everything is still working.

**Comment by dpwood** [comment]
The problem as described by this PR doesn't fit reports of #2579.

It's not a problem of intense parallel activity. The symptom I've witnessed is sustained high CPU and increasing memory consumption until it runs out of memory and crashes.

**Comment by danielbrunt57** [comment]
But the helper function has never been altered so I'm trying to understand what external change has caused the helper to now be in an infinite loop.  If you don't know that's fine as processing a copy of the list when the original list is going to be changed by the process is the correct way otherwise you usually end up shooting yourself in the foot!

**Comment by dpwood** [comment]
> But the helper function has never been altered so I'm trying to understand what external change has caused the helper to now be in an infinite loop. If you don't know that's fine as processing a copy of the list when the original list is going to be changed by the process is the correct way otherwise you usually end up shooting yourself in the foot!

It was altered:

https://github.com/alandtse/alexa_media_player/commit/6f9a92de6fda7fd5361e15ae719c11092eaf7228#diff-4e8bc1554181c711fd75da33fbb2b0e7a8a1abd5096c723a1d3baa9f3c10543c

**Comment by alandtse** [comment]
Went with #2580

**Pull request closed without merging**